src/libsyntax/json.rs

```diff
@@ -81,6 +81,14 @@ impl Emitter for JsonEmitter {
             panic!("failed to print diagnostics: {:?}", e);
         }
     }
+
+    fn emit_directive(&mut self, directive: String) {
+        let data = Directive { directive };
+        let result = writeln!(&mut self.dst, "{}", as_pretty_json(&data));
```

Comment:
Most tools processing the rustc command line expect one json object per line. And they do this by looking for lines starting with `{`. This is also what causes the travis failure.

We have the `pretty` field on `JsonEmitter` to distinguish between user-readable json and tool-processable json (see `emit_diagnostic` above).